wdl2cwl/main.py

```diff
@@ -277,99 +282,117 @@ def load_wdl_workflow(self, obj: WDL.Tree.Workflow) -> cwl.Workflow:
+        requirements = [cwl.ScatterFeatureRequirement()] if is_scatter_present else None
 
         return cwl.Workflow(
             id=wf_name,
             cwlVersion="v1.2",
             doc=wf_description,
+            requirements=requirements,
             inputs=inputs,
             steps=wf_steps,
             outputs=outputs,
         )
 
+    def get_workflow_scatter(self, scatter: WDL.Tree.Scatter) -> List[cwl.WorkflowStep]:
+        """Get the CWL Workflow Step equivalent of a list of WDL Scatter Object."""
+        scatter_steps: List[cwl.WorkflowStep] = []
+        for body_obj in scatter.body:
+            if isinstance(body_obj, WDL.Tree.Call):
+                wp_step = self.get_workflow_call(body_obj)
+            wp_step.scatter = self.scatter_names.pop()
+            scatter_steps.append(wp_step)
+        return scatter_steps
+
+    def get_workflow_call(self, call: WDL.Tree.Call) -> cwl.WorkflowStep:
+        """Get the CWL Workflow Step equivalent of a WDL Call Object."""
+        callee = call.callee
+        if callee is None:
+            raise ConversionException("callee is None. This should not be possible.")
+        cwl_callee_inputs = self.get_cwl_task_inputs(callee.inputs)
+        call_inputs = call.inputs
+        inputs_from_call: Dict[str, Tuple[str, Dict[str, Any]]] = {}
+        input_defaults = set()
+        if call_inputs:
+            scatter_handled = False
+            for key, value in call_inputs.items():
+                with WDLSourceLine(value, ConversionException):
+                    if not isinstance(value, (WDL.Expr.Get, WDL.Expr.Apply)):
+                        input_defaults.add(key)
+                    if (
+                        isinstance(value, WDL.Expr.Apply)
+                        and value.function_name == "select_all"
+                        and isinstance(value.arguments[0], WDL.Expr.Array)
+                        and isinstance(
+                            value.arguments[0].items[0],
+                            (WDL.Expr.Get, WDL.Expr.String),
+                        )
+                    ):
+                        input_expr = self.get_step_input_expr(
+                            value.arguments[0].items[0]
+                        )
+                        inputs_from_call[key] = (
+                            input_expr.replace(".", "/"),
+                            {
+                                "pickValue": "all_non_null",
+                                "valueFrom": "$([self])",
+                            },
+                        )
+                    else:
+                        input_expr = self.get_step_input_expr(value)  # type: ignore[arg-type]
+                        inputs_from_call[key] = (
+                            input_expr.replace(".", "/"),
+                            {},
+                        )
+                    if self.scatter_names and not scatter_handled:
+                        self.scatter_names[-1] = key
```

Comment:
When a scatter input is encountered and the scatter's source has been appended to the scatter_names stack, it us updated with the destination which is the key. 

scatter_handled is updated to True so as to prevent the scatter_names from been updated when the call has more than one inputs. 

This implementation only handles one scatter input per scatter call.

Reply:
Ah, ok, I think I understood how `scatter_names` is used. I think here we could empty the stack? Unless there's further need for `scatter_names`, or maybe make a copy at the beginning of the function, then empty as we process it. That would remove the need for the `scatter_handled` variable and make it easier to comprehend, I think.

Reply:
If the parent of the call is a scatter object then the `scatter_names` stack will still be used there. The last name entered into the `scatter_names` stack is popped at get_workflow_scatter.